lib/generator.js

```diff
@@ -288,6 +288,15 @@ class Generator {
    * @return {Promise}
    */
   generateDirectoryStructure(asyncapiDocument) {
+    const fileNamesForSeperation = {
+      channel: asyncapiDocument.channels(),
+      message: asyncapiDocument.components().messages(),
+      securityScheme: asyncapiDocument.components().securitySchemes(),
+      scheme: asyncapiDocument.components().schemas(),
+      parameter: asyncapiDocument.components().parameters(),
+      messageTrait: asyncapiDocument.components().messageTraits()
+    };
```

Comment:
I understand the temptation of adding more "components" but we should be careful. Let's add things as people request them instead of adding them because they are easy. Think that we'll have to maintain it and we may end up maintaining something that is not useful to anyone. And then removing it will mean that we'd have to release a major version :)

Reply:
Yes that's true, but on the other hand not having the feature right from the start might be frustrating to some if they use it for the first time, and since it is using the same logic being used it might be better off just adding them now 😄 But I can see both points of view,  let me know if I should delete it

Reply:
Hehe! That's true. If you can think of a good use case for message traits to be in a separate, go for it. Otherwise, let's delete it :)

Reply:
Hmm, one use case I can think of is inheritance, multiple messages using the same trait and inherits it instead of having duplicated code (same with operation traits). However not quite sure people will care if they are generated anyway by the template 🤔

Reply:
Just gonna remove it for now, easy to add later :)